modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_folder.jsp

```diff
@@ -486,34 +490,42 @@ renderResponse.setTitle(headerTitle);
 
 			searchContainer.deleteRow(tr, link.getAttribute('data-rowId'));
 
-			A.one(
-				'#<portlet:namespace />defaultFileEntryTypeId-' +
+			var option = document.getElementById(
+				'<portlet:namespace />defaultFileEntryTypeId-' +
 					link.getAttribute('data-rowId')
-			).remove();
+			);
+
+			option.parentElement.removeChild(option);
 
 			<portlet:namespace />documentTypesChanged = true;
 
-			var select = A.one(
-				'#<%= liferayPortletResponse.getNamespace() + "workflowDefinition" + DLFileEntryTypeConstants.FILE_ENTRY_TYPE_ID_ALL %>'
+			var select = document.getElementById(
+				'<%= liferayPortletResponse.getNamespace() + "workflowDefinition" + DLFileEntryTypeConstants.FILE_ENTRY_TYPE_ID_ALL %>'
 			);
 
-			var selectContainer = A.one(
-				'#<portlet:namespace />restrictionTypeWorkflow'
+			var selectContainer = document.getElementById(
+				'<portlet:namespace />restrictionTypeWorkflow'
 			);
 
-			var fileEntryTypesCount = select.get('children').size();
+			var fileEntryTypesCount = select.children.length;
 
 			if (fileEntryTypesCount == 0) {
-				selectContainer.hide();
+				selectContainer.classList.add('hide');
+				selectContainer.setAttribute('hidden', 'hidden');
+				selectContainer.style.display = 'none';
 
-				var restrictionTypeWorkflow = A.one(
-					'#<portlet:namespace />restrictionTypeWorkflow'
+				var restrictionTypeWorkflow = document.getElementById(
+					'<portlet:namespace />restrictionTypeWorkflow'
 				);
 
-				restrictionTypeWorkflow.show();
+				restrictionTypeWorkflow.classList.remove('hide');
+				restrictionTypeWorkflow.removeAttribute('hidden');
+				restrictionTypeWorkflow.style.display = '';
 			}
 			else {
-				selectContainer.show();
+				selectContainer.classList.remove('hide');
+				selectContainer.removeAttribute('hidden');
+				selectContainer.style.display = '';
```

Comment:
Given that this is a repetitive multi-step operation I would have declared a couple of little local helper functions inside the closure:

```js
function show(element) {
    element.classList.remove('hide');
    element.removeAttribute('hidden');
    element.style.display = '';
}

function hide(element) {
    element.classList.add('hide');
    element.setAttribute('hidden', 'hidden');
    element.style.display = 'none';
}
```

(No need to make this change; just pointing out a possible refactoring.)

Reply:
You're right, and when replacing AUI's `hide` and `show`, I've had to do this several times in various modules. 

I don't know if this should be in our "core" utils (which I didn't want to "pollute" until we spoke about it), but it's used quite a lot.

Reply:
I know that for some people, keeping things "DRY" is a religion, and that means pushing stuff into a core module if at all possible. Personally, I wouldn't take that step because:

1. In order to be truly DRY, you have to use it everywhere, which means auditing the code base, figuring out how to import it everywhere it is needed (which may be complicated because some of the uses will be in JSPs files and you'll be back in that same territory where you've spent the last few weeks of analyzing dependency and ordering issues etc), then you need to educate people to use the new tool (write documentation, post an announcement, encourage use), and if you want to be sure that they use it you need to write a lint too that catches usages of the "wrong API" (may be nontrivial). If you extract but don't do all the rest, you'll just be contributing to [the "15 competing standards" problem](https://xkcd.com/927/), where we wind up with one more way of doing the same thing.
2. The logic is so simple that extracting it out is akin to making an [is-odd](https://www.npmjs.com/package/is-odd) package (half-a-million weekly downloads, BTW). Abstraction is good when it enables you to address more complex problems by making things more intellectually manageable. In this case, the abstraction does nothing of the sort; it just declutters things by eliminating some repeated trivial operations.
3. I don't think we should be in the business of making it too easy to manipulate the DOM from inside JSP files. On the contrary, we need to be investing in infrastructure improvements that make it easier to use better tools (eg. React). When we can't do that due to performance reasons or other concerns, it's our job as an infra team to work on eliminating the root causes of those problems until we get to the point where people can develop on Liferay with one simple mental model ("I just use React") in a systematic way, and we take care of the details.

Reply:
You beat me to it, @wincent!

![moral_support](https://user-images.githubusercontent.com/905006/85278424-3de09580-b485-11ea-84d3-2035883140ce.gif)